app/src/main/java/com/mathewsachin/fategrandautomata/accessibility/ScriptRunnerUserInterface.kt

```diff
@@ -33,6 +33,29 @@ class ScriptRunnerUserInterface(val Service: ScriptRunnerService) {
 
     val windowManager = Service.getSystemService(Context.WINDOW_SERVICE) as WindowManager
 
+    private val metrics: DisplayMetrics get() {
+        val res = DisplayMetrics()
+
+        windowManager.defaultDisplay.getRealMetrics(res)
+
+        return res
+    }
+
+    val mediaProjectionMetrics: DisplayMetrics get() {
+        val res = metrics
+
+        // Retrieve images in Landscape
+        if (res.heightPixels > res.widthPixels) {
+            res.let {
```

Comment:
Why do you even use `.let`?

Also, I searched for swapping variables in Kotlin and found this:
```
res.widthPixels = res.heightPixels.also { res.heightPixels = res.widthPixels}
```

Reply:
let, also, run, apply can be used to manipulate an item without referring to it again and again by name. There are some differences like you refer the object using `it` or `this`

Swapping using also doesn't look good to me, so didn't use that. Maybe there's some other way.

Reply:
In my opinion, `it` is pretty ambiguous, it could mean anything, and `let` is only useful when you don't want to assign something to a variable. In this case using `res` is the same amount of effort as using `it`.

Reply:
Umm, `it` isn't ambiguous. It is a convention in Kotlin. If there is only one parameter to a lambda, it's name by default is `it`.
Yeah, `run` should probably be used here in place of `let`.
The benefit of this approach is that `temp` doesn't leak out of that scope.
Another benefit I can think is if we rename `res` later to something else, maybe a longer more descriptive name, the amount of change required will be minimal.

Reply:
`temp` is also only visible inside of the `if`. I still find using `res` better, but you can merge if you want to keep `let`.

Reply:
I want to keep `let` 😅